### PR TITLE
Finalize the Download Command changes

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This PR finalizes #126, which includes #194 from @nightman68 and #195 from me.

- [x] Refactor the code
- [x] Switch `unsecureAuth` to the boolean flag, update docs
- [x] Make `auth` field case-insensitive
- [x] Add unit tests for the authentication functionality
- [x] Update configuration samples

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@nightman68 Ready for review. CC @reviewbybees just in case

File: examples/sample-allOptions.xml [hunk 1]
```diff
@@ -251,18 +251,20 @@ SECTION: Environment setup
...
-            auth="basic" unsecureAuth=“enabled”
+            auth="basic" unsecureAuth=“true”
             username="aUser" password=“aPassw0rd" />
 
   Secure Basic authentication via HTTPS:
   <download from="https://example.com/some.dat" to="%BASE%\some.dat"
             auth="basic" username="aUser" password="aPassw0rd" />
 
-  Secure authentication when the target server and the client are members of domain:
+  Secure authentication when the target server and the client are members of the same domain:
```
Comment (nightman68) [inline examples/sample-allOptions.xml]: The membership in "the same" domain is not normally not the constraint. When a forest trust is defined correctly it is also working.

Comment (oleg-nenashev) [inline examples/sample-allOptions.xml]: Right. "of domain" does not seem the best wording as well. Could it be there a better clarification text?

Comment (nightman68) [inline examples/sample-allOptions.xml]: Right .-) My proposal:
Secure authentication when the target server and the client are members of the same domain or the server domain and the client domain belong to the same forest with a trust.

Comment (oleg-nenashev) [inline examples/sample-allOptions.xml]: updated

Comment (nightman68) [inline examples/sample-allOptions.xml]: 👍

**Comment by oleg-nenashev** [comment]
Seems it is ready to go

**Pull request merged**